### PR TITLE
octave: provides fltk with opengl, adds qscintilla

### LINKED ISSUE #21650: Installation issue: octave
When building octave 6.1.0 with fltk support, there is no build error, but the resulting build does not use fltk.

### Steps to reproduce the issue

```console
$ spack install octave@6.1.0  +fftw  +fontconfig +freetype +gnuplot +curl +fltk
...
```
Lack of fltk functionality can be verified by running the octave-6.1.0 binary, and checking the value of available_graphics_toolkits(). It will be found that only the gnuplot toolkit is available, i.e. the fltk support was not linked into the binary.

### Information on your system

* **Spack:** 0.16.0-1262-0724467ad2
* **Python:** 3.8.5
* **Platform:** linux-ubuntu20.04-skylake
* **Concretizer:** original

### Additional information

<https://octave.1599824.n4.nabble.com/Octave-3-4-2-with-fltk-support-td3627783.html> indicates that mesa support must be available at link time in order for fltk support to be functional. Probably mesa should be added as a dependency for fltk.

- [ x] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by cessenat**
fixes #21650 

Solves issue #21650 where +fltk did not bring the package in octave using available_graphics_toolkits() 
Adds +qscintilla option.

### CODE REVIEW

**Review by alalazo — APPROVED** [review]
@mtmiller

**Comment by cessenat** [comment]
Hello, is there a reason you did not merge it to develop ? - other than time of course !
Thanks anyway.
BTW I have a few PR with no reviewer...

**Comment by mtmiller** [comment]
Looks sensible to me.

Do the spack maintainers prefer dependencies to be combined, e.g. `fltk+gl` for a given option, or separated like this PR does?

File: var/spack/repos/builtin/packages/octave/package.py
Comment (mtmiller) [inline var/spack/repos/builtin/packages/octave/package.py:93]: What's the difference between `gl` here and `opengl` below?

Comment (cessenat) [reply var/spack/repos/builtin/packages/octave/package.py:93]: I just saw many other package depend on 'gl' and none on 'opengl'. But this is not an explanation, sorry.

Comment (mtmiller) [reply var/spack/repos/builtin/packages/octave/package.py:93]: I think it may be that `gl` is a package, but `opengl` in `qt+opengl` is a build option.

Maybe the fltk package should also support a `+opengl` option?

**Comment by cessenat** [comment]
> Do the spack maintainers prefer dependencies to be combined, e.g. `fltk+gl` for a given option, or separated like this PR does?

I'm not a maintainer of course but as found out in issue  #21650 octave needs OpenGL for the fltk option to work ; this may indicate that we ought to add a variant for fltk, but I'm not sure.

File: var/spack/repos/builtin/packages/octave/package.py
Comment (cessenat) [reply var/spack/repos/builtin/packages/octave/package.py:93]: Exactly for both. But maybe only.

Comment (alalazo) [reply var/spack/repos/builtin/packages/octave/package.py:93]: `gl` is a virtual dependency (like e.g. `mpi` or `lapack`) with many implementations:
```console
$ spack providers gl
gl:
mesa    opengl       opengl@1.1:  opengl@1.3:  opengl@1.5:  opengl@2.1:  opengl@3.1:  opengl@3.3:  opengl@4.2:  opengl@4.4:
mesa18  opengl@1.0:  opengl@1.2:  opengl@1.4:  opengl@2.0:  opengl@3.0:  opengl@3.2:  opengl@4.1:  opengl@4.3:  opengl@4.5:
```
If a package just depends on `gl` it depends on any other package providing the API or functionality. If it depends on `opengl` it depends on that specific implementation (as if it dependend on `openblas` or `mpich`). There are also variants with the same name to activate dependencies in case it makes sense to have them conditional, but that's a separate topic.

**Comment by alalazo** [comment]
> Hello, is there a reason you did not merge it to develop ?

Yes, the package has a maintainer so I was waiting some time for @mtmiller to have a look at the PR :slightly_smiling_face:

**Comment by cessenat** [comment]
> > Hello, is there a reason you did not merge it to develop ?
> 
> Yes, the package has a maintainer so I was waiting some time for @mtmiller to have a look at the PR

All right, naturally. Thanks to you both

**Pull request merged**
